src/procedures/proc_functions.c

```
@@ -0,0 +1,141 @@
+	raxSeek(&pdata->iter, "^", NULL, 0);
+	pdata->output = array_new(SIValue, 3);
+	_process_yield(pdata, yield);
+
+	ctx->privateData = pdata;
+	return PROCEDURE_OK;
+}
+
+
+// promote the rax iterator to the next function and return its name and mode.
+SIValue *Proc_FunctionsStep
+(
```

cppcheck

[error] Memory leak

(memleak)